### PR TITLE
DDF for SONOFF motion sensor SNZB-03P

### PR DESCRIPTION
**Pull request by Smanar**
By @febalci 
Product name : SONOFF SNZB-03P Motion Sensor [Link](https://sonoff.tech/product/gateway-and-sensors/snzb-03p/)
Manufacturer : SONOFF
Model identifier : SNZB-03P
Device type to add : Sensor

See https://github.com/dresden-elektronik/deconz-rest-plugin/issues/7821

### CODE REVIEW

File: devices/sonoff/snzb-03p_motion_sensor.json [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+      "bind": "unicast",
+      "src.ep": 1,
+      "cl": "0x0001",
+      "report": [
+        {
+          "at": "0x0021",
+          "dt": "0x20",
+			"min": 3600,
+			"max": 7200,
+			"change": "0x00000002"
```
Comment (SwoopX) [inline devices/sonoff/snzb-03p_motion_sensor.json:187]: You might want to ament the indentation.

File: devices/sonoff/snzb-03p_motion_sensor.json [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            "cl": "0x0001",
+            "ep": 1,
+            "eval": "Item.val = Attr.val / 2;",
+            "fn": "zcl:attr"
+          },
+          "default": 0
+        },
+        {
+          "name": "config/delay",
+          "refresh.interval": 3600,
```
Comment (SwoopX) [inline devices/sonoff/snzb-03p_motion_sensor.json:79]: You might want to doublecheck the refresh interval, as reporting is significantly lower

File: devices/sonoff/snzb-03p_motion_sensor.json [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        },
+        {
+          "name": "attr/type"
+        },
+        {
+          "name": "attr/uniqueid"
+        },
+        {
+          "name": "config/battery",
+          "awake": true,
```
Comment (SwoopX) [inline devices/sonoff/snzb-03p_motion_sensor.json:60]: Are you sure the device is only awake when it sends battery reports?

File: devices/sonoff/snzb-03p_motion_sensor.json [hunk 4]
```diff
@@ -0,0 +1,192 @@
+{
+  "schema": "devcap1.schema.json",
+  "manufacturername": "eWeLink",
+  "modelid": "SNZB-03P",
+  "vendor": "SONOFF",
+  "product": "SNZB-03P",
+  "sleeper": false,
```
Comment (SwoopX) [inline devices/sonoff/snzb-03p_motion_sensor.json:7]: Sure? 🙂

File: devices/sonoff/snzb-03p_motion_sensor.json [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+      ],
+      "fingerprint": {
+        "profile": "0x0104",
+        "device": "0x0107",
+        "endpoint": "0x01",
+        "in": [
+          "0x0000",
+          "0x0001",
+          "0x0406",
+          "0x0500",
```
Comment (SwoopX) [inline devices/sonoff/snzb-03p_motion_sensor.json:26]: Just wondering if the device actually uses the IAS Zone cluster or if it has any effect. I'd conclude the device doesn't require any enrollment to work?

**Review by SwoopX — COMMENTED** [review]
Just a few quick things to check/amend, nothing serious 👍

File: devices/sonoff/snzb-03p_motion_sensor.json (hunk 2 above)
Comment (Smanar) [reply devices/sonoff/snzb-03p_motion_sensor.json:79]: In a perfect world, "refresh.interval" is never used, it's just a security for me, the setting is only changed 2 times in a year.
Have just copy/paste other DDF (working I hope)

File: devices/sonoff/snzb-03p_motion_sensor.json (hunk 1 above)
Comment (Smanar) [reply devices/sonoff/snzb-03p_motion_sensor.json:187]: corrected

File: devices/sonoff/snzb-03p_motion_sensor.json (hunk 4 above)
Comment (Smanar) [reply devices/sonoff/snzb-03p_motion_sensor.json:7]: Changed something

File: devices/sonoff/snzb-03p_motion_sensor.json (hunk 5 above)
Comment (Smanar) [reply devices/sonoff/snzb-03p_motion_sensor.json:26]: I think no
And there is not config/enrollemetn and config/pending.

Removed.

File: devices/sonoff/snzb-03p_motion_sensor.json (hunk 4 above)
Comment (SwoopX) [reply devices/sonoff/snzb-03p_motion_sensor.json:7]: Sorry, my bad. I meant `"sleeper": false,`, as I consider this as rather unusual for a battery powered device.

File: devices/sonoff/snzb-03p_motion_sensor.json (hunk 2 above)
Comment (SwoopX) [reply devices/sonoff/snzb-03p_motion_sensor.json:77]: Yeah, absolutely agree. I guess it doesn't matter anyway if 1-2 reports get lost, the poll is very unlikely to happen at all.

File: devices/sonoff/snzb-03p_motion_sensor.json (hunk 4 above)
Comment (Smanar) [reply devices/sonoff/snzb-03p_motion_sensor.json]: Arf, right, have see the device, no power, and on the GUI it's a end device, no reason for the device was not a slepper

**Comment by saxn-paule** [comment]
The SNZB-03 PIR Motion sensor is detected as water leak sensor in Phoscon 2.30.2

**Comment by Smanar** [comment]
You are sure it's not from phoscon ?
Can take a look direclty in the API using Phoscon/help/API Information/sensor ?

**Comment by saxn-paule** [comment]
image

**Comment by Smanar** [comment]
This PR is for the SNZB-03P, so not the same than you.
Perhaps https://github.com/dresden-elektronik/deconz-rest-plugin/issues/8301

**Comment by saxn-paule** [comment]
> This PR is for the SNZB-03P, so not the same than you. Perhaps #8301

#8301 was created by me 😉

**Comment by Smanar** [comment]
Lol, too much crossed issue in same time for me.

**Pull request merged**